Fix #3332: Created a BUILD.bazel file in domain/audio

yash10019coder opened the pull request:
<!-- READ ME FIRST: Please fill in the explanation section below and check off every point from the Essential Checklist! -->
## Explanation
<!--
  - Explain what your PR does. If this PR fixes an existing bug, please include
  - "Fixes #bugnum:" in the explanation so that GitHub can auto-close the issue
  - when this PR is merged.
  -->
Fixes #3332 
Created the bazel file in audio and also added targets to the needed targets.
## Essential Checklist
<!-- Please tick the relevant boxes by putting an "x" in them. -->
- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only
<!-- Delete these section if this PR does not include UI-related changes. -->
N/A -- this PR is only changing the Bazel build graph and doesn't affect any user flows.

yash10019coder committed:
created the targets in the audio/bazel and also added the targets to the needed targets

yash10019coder committed:
removed the commented code in audio/bazel

yash10019coder reviewed (commented):
Hi @BenHenning created a PR PTAL

yash10019coder committed:
reformatted domain/bazel

BenHenning commented:
Thanks @yash10019coder. Apologies, will need to look at this tomorrow.

BenHenning commented on domain/BUILD.bazel:
```diff
@@ -100,6 +100,8 @@ kt_android_library(
         "//data/src/main/java/org/oppia/android/data/backends/gae/api",
         "//data/src/main/java/org/oppia/android/data/persistence:cache_store",
+        "//domain/src/main/java/org/oppia/android/domain/audio:audio_player_controller",
+        "//domain/src/main/java/org/oppia/android/domain/audio:cellular_audio_dialog_controller",
```
Do any domain classes actually depend on these, or is it only UI? Prefer importing it in the UI layer if the latter since we shouldn't be relying on implicit dependencies this way.

BenHenning commented on domain/src/main/java/org/oppia/android/domain/audio/BUILD.bazel:
```diff
@@ -0,0 +1,40 @@
+"""
+Package for audio controlling
```
```suggestion
Domain services & definitions corresponding to managing and playing audio tracks.
```

Bit clearer on what "controlling" means, and is consistent with another modularized domain subpackage.

BenHenning reviewed (changes requested):
Thanks @yash10019coder. Just had a couple of comments, otherwise the PR looks quite good to me.

Regarding the PR itself, three things:
1. Could you please update the title? It's currently cut-off. Titles should generally be concise and specific as to the high-level thing the PR is accomplishing.
2. Could you please move the explanation portion in your PR description to be under the 'explanation' section? 
3. You can (& should) remove the bits under "For UI-specific PRs only" and describe why this PR doesn't need that (e.g. "N/A -- this PR is only changing the Bazel build graph and doesn't affect any user flows"). In general, please make sure to follow the instructions in the PR description to ensure that all the needed pieces are included. This really helps both reviewers and future people who come across the PR after it gets merged.

yash10019coder committed:
Update domain/src/main/java/org/oppia/android/domain/audio/BUILD.bazel

Co-authored-by: Ben Henning <henning.benmax@gmail.com>

yash10019coder committed:
Merge branch 'oppia:develop' into create-build-bazel-for-domain-audio

yash10019coder committed:
added the test dependencies to app/bazel of audio_player_controller

yash10019coder reviewed (commented):
@BenHenning Please PTAL

yash10019coder committed:
added deps in app/bazel

yash10019coder committed:
Merge branch 'create-build-bazel-for-domain-audio' of github.com:yash10019coder/oppia-android into create-build-bazel-for-domain-audio

yash10019coder committed:
Revert "added deps in app/bazel"

This reverts commit 7532aadbd579e80b14cab34ce014c3b5328511a3.

yash10019coder committed:
Revert "added the test dependencies to app/bazel of audio_player_controller"

This reverts commit 4b29c9e878ba835aaf72741848c434563aede012.

yash10019coder committed:
added dependencies in app/bazel and domain/bazel of audio/bazel

yash10019coder committed:
reformatted domain/bazel

yash10019coder reviewed (commented):
Hi @BenHenning added the dependencies and removed the old dependencies from domian/bazel PTAL

BenHenning reviewed (approved):
Thanks @yash10019coder! The PR LGTM.

Two things:
1. I updated your PR description since you didn't address point (3) in my earlier comment. Please see how I changed it for a reference on future PRs--copying that approach will help streamline future PRs that you send.
2. Please remember to reply to comment threads directly when they're addressed

yash10019coder commented:
Thanks a lot, @BenHenning for merging this PR 😄

yash10019coder commented:
> Thanks @yash10019coder! The PR LGTM.
> 
> Two things:
> 
> 1. I updated your PR description since you didn't address point (3) in my earlier comment. Please see how I changed it for a reference on future PRs--copying that approach will help streamline future PRs that you send.
> 2. Please remember to reply to comment threads directly when they're addressed

apologies for not replying I forgot I will always reply in the future for all the comments
thanks

BenHenning commented:
> > Thanks @yash10019coder! The PR LGTM.
> > Two things:
> > 
> > 1. I updated your PR description since you didn't address point (3) in my earlier comment. Please see how I changed it for a reference on future PRs--copying that approach will help streamline future PRs that you send.
> > 2. Please remember to reply to comment threads directly when they're addressed
> 
> apologies for not replying I forgot I will always reply in the future for all the comments thanks

Thanks @yash10019coder, I appreciate it. :)

yash10019coder commented:
Thanks for merging the PR

On Tue, Nov 2, 2021, 5:50 AM Ben Henning ***@***.***> wrote:

> Thanks @yash10019coder <https://github.com/yash10019coder>! The PR LGTM.
> Two things:
>
>    1. I updated your PR description since you didn't address point (3) in
>    my earlier comment. Please see how I changed it for a reference on future
>    PRs--copying that approach will help streamline future PRs that you send.
>    2. Please remember to reply to comment threads directly when they're
>    addressed
>
> apologies for not replying I forgot I will always reply in the future for
> all the comments thanks
>
> Thanks @yash10019coder <https://github.com/yash10019coder>, I appreciate
> it. :)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/oppia/oppia-android/pull/3967#issuecomment-956976269>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ASG5QUNQLP2NAYHFQLEGAMTUJ4VDDANCNFSM5GTVKHAA>
> .
>
